### ISSUE TITLE
Can't read XML files with colon ":" in path from Azure Data Lake Storage Gen2

### ISSUE THREAD

**Issue by MarwanDebbiche** [databricks/spark-xml#512]
I am trying to read an xml file from an Azure Data Lake Storage Gen2 using the python API, but I get an URISyntaxException when the file path contains colons:

```python
input_path = "abfss://container@account.dfs.core.windows.net/books_2021-01-11-09:23:11"
df = spark.read.format("xml").options(rowTag="book", inferSchema=True).load(input_path)
```

```
IllegalArgumentException: java.net.URISyntaxException: Relative path in absolute URI: books_2021-01-1109:23:11
```

Everything works fine if I rename the file to remove the colons.

I tried reading CSV or parquet files with colons in their paths and it works fine.

I also tried escaping the file name but it then raises an InvalidInputException:

```python
input_path = "abfss://container@account.dfs.core.windows.net/books_2021-01-1109%3A23%3A11"
df = spark.read.format("xml").options(rowTag="book", inferSchema=True).load(input_path)
```

```
org.apache.hadoop.mapreduce.lib.input.InvalidInputException: Input path does not exist
```

Is there a way to read XML files whose paths contains colons?

Thank you

**Comment by srowen** [comment]
Huh. Yeah I would have said the colon needs to be escaped, but that doesn't work either as it is not un-escaped then by the Hadoop libs. I can reproduce this. So far I do not see a workaround; other escapes didn't work. I'm not sure why CSV/Parquet would be different as they should all go through Hadoop's Path class and InputFormats.

**Comment by MarwanDebbiche** [comment]
Thanks for replying @srowen.

I have indeed read [here](https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/site/markdown/filesystem/model.md) that colons should not be included in hdfs path. I guess it's the same for Azure Data Lake Storage Gen2.

So I don't know why it is working with spark's CSV and parquet built-in reader, but it is.

Anyway, in the end we decided to remove colons from file paths.
